src/test/java/uk/gov/pay/directdebit/pact/PublicApiContractTest.java

```diff
@@ -0,0 +1,43 @@
+package uk.gov.pay.directdebit.pact;
+
+import au.com.dius.pact.provider.junit.PactRunner;
+import au.com.dius.pact.provider.junit.Provider;
+import au.com.dius.pact.provider.junit.State;
+import au.com.dius.pact.provider.junit.loader.PactBroker;
+import au.com.dius.pact.provider.junit.loader.PactBrokerAuth;
+import au.com.dius.pact.provider.junit.target.HttpTarget;
+import au.com.dius.pact.provider.junit.target.Target;
+import au.com.dius.pact.provider.junit.target.TestTarget;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.runner.RunWith;
+import uk.gov.pay.directdebit.junit.DropwizardAppWithPostgresRule;
+import uk.gov.pay.directdebit.payments.fixtures.GatewayAccountFixture;
+
+import java.util.Map;
+
+@RunWith(PactRunner.class)
+@Provider("direct-debit-connector")
+@PactBroker(protocol = "https", host = "pact-broker-test.cloudapps.digital", port = "443", tags = {"${PACT_CONSUMER_TAG}"},
+        authentication = @PactBrokerAuth(username = "${PACT_BROKER_USERNAME}", password = "${PACT_BROKER_PASSWORD}"))
+//@PactFolder("pacts") <-- this is useful for testing pacts locally
+public class PublicApiContractTest {
+
+    @ClassRule
+    public static DropwizardAppWithPostgresRule app = new DropwizardAppWithPostgresRule();
+    
+    @TestTarget
+    public static Target target;
+
+    private GatewayAccountFixture testGatewayAccount = GatewayAccountFixture.aGatewayAccountFixture();
+
+    @BeforeClass
+    public static void setUpService() {
+        target = new HttpTarget(app.getLocalPort());
+    }
+    
+    @State("a gateway account with external id exists")
```

Comment:
I think we should specify the id in the provider state string (I realise this would need changes in both publicapi and here). The more specific and informative we can make these state strings, the less confused we'll be, I hope

Reply:
My view is that using the `Map<String, String> params` would be better as this would allow code reuse. If you have a look at https://github.com/alphagov/pay-publicapi/blob/BAU-add-pact-testing/src/test/resources/pacts/publicapi-publicauth-direct-debit.json#L11 for example it defines a state "A bearer token exists" that provides a "type" and "id". I can easily see another pact being created that needs the same but with a different "type" (CARD for example). This means we'd only have to define one `@State` method in the provider test.

Reply:
oh that's cool, I didn't know you could do that

Reply:
yeah one of the ways pact-jvm is better than the JS one :)

Reply:
👍